OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -198,61 +198,83 @@ internal static void UpgradeWeaponRules(int engineVersion, ref List<MiniYamlNode
 			{
 				// Add rules here
-				UpgradeTileset(engineVersion, ref node.Value.Nodes, node, depth + 1);
+				UpgradeTileset(modData, engineVersion, ref node.Value.Nodes, node, depth + 1);
 			}
 		}
 
-		internal static void UpgradeCursors(int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
+		internal static void UpgradeCursors(ModData modData, int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
 		{
 			foreach (var node in nodes)
 			{
 				// Add rules here
-				UpgradeCursors(engineVersion, ref node.Value.Nodes, node, depth + 1);
+				UpgradeCursors(modData, engineVersion, ref node.Value.Nodes, node, depth + 1);
 			}
 		}
 
-		internal static void UpgradePlayers(int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
+		internal static void UpgradePlayers(ModData modData, int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
 		{
 			foreach (var node in nodes)
 			{
 				// Add rules here
-				UpgradePlayers(engineVersion, ref node.Value.Nodes, node, depth + 1);
+				UpgradePlayers(modData, engineVersion, ref node.Value.Nodes, node, depth + 1);
 			}
 		}
 
-		internal static void UpgradeChromeMetrics(int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
+		internal static void UpgradeChromeMetrics(ModData modData, int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
 		{
 			foreach (var node in nodes)
 			{
 				// Add rules here
-				UpgradeChromeMetrics(engineVersion, ref node.Value.Nodes, node, depth + 1);
+				UpgradeChromeMetrics(modData, engineVersion, ref node.Value.Nodes, node, depth + 1);
 			}
 		}
 
-		internal static void UpgradeChromeLayout(int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
+		internal static void UpgradeChromeLayout(ModData modData, int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
 		{
 			foreach (var node in nodes)
 			{
 				// Add rules here
-				UpgradeChromeLayout(engineVersion, ref node.Value.Nodes, node, depth + 1);
+				UpgradeChromeLayout(modData, engineVersion, ref node.Value.Nodes, node, depth + 1);
 			}
 		}
 
-		internal static void UpgradeActors(int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
+		internal static void ModifyCPos(ref string input, CVec vector)
+		{
+			var oldCPos = FieldLoader.GetValue<CPos>("(value)", input);
+			var newCPos = oldCPos + vector;
+			input = newCPos.ToString();
+		}
+
+		internal static void UpgradeActors(ModData modData, int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth)
 		{
 			foreach (var node in nodes)
 			{
-				// Add rules here
-				UpgradeActors(engineVersion, ref node.Value.Nodes, node, depth + 1);
+				// Fix RA building footprints to not use _ when it's not necessary
+				if (engineVersion < 20160530 && modData.Manifest.Mod.Id == "ra" && depth == 2)
```

Comment:
> These need to be depth == 1 if you want the rule to run

Hm... are you sure? I thought using `parent` for the actor node (which should be depth 1) and `node` for the `Location` (should be depth 2) would be correct.